src/librustc_lint/types.rs

```diff
@@ -83,8 +83,8 @@ fn lint_overflowing_range_endpoint<'a, 'tcx>(
                     // We need to preserve the literal's suffix,
                     // as it may determine typing information.
                     let suffix = match lit.node {
-                        LitKind::Int(_, LitIntType::Signed(s)) => format!("{}", s.name_str()),
-                        LitKind::Int(_, LitIntType::Unsigned(s)) => format!("{}", s.name_str()),
+                        LitKind::Int(_, LitIntType::Signed(s)) => s.name_str().to_owned(),
+                        LitKind::Int(_, LitIntType::Unsigned(s)) => s.name_str().to_owned(),
```

Comment:
Any reason these are `to_owned`?

Reply:
the line below was already using `to_owned()` so I thought I would go with it. :shrug:

Reply:
I switched all of them to to_string() now.